ding/reward_model/curiosity_reward_model.py

```diff
@@ -0,0 +1,185 @@
+from typing import Union, Tuple
+from easydict import EasyDict
+
+import random
+import torch
+import torch.nn as nn
+import torch.optim as optim
+import torch.nn.functional as F
+
+from ding.utils import SequenceType, REWARD_MODEL_REGISTRY
+from ding.model import FCEncoder, ConvEncoder
+from .base_reward_model import BaseRewardModel
+
+
+def collect_states(iterator):
+    states = []
+    next_states = []
+    actions = []
+    for item in iterator:
+        state = item['obs']
+        next_state = item['next_obs']
+        action = item['action']
+        states.append(state)
+        next_states.append(next_state)
+        actions.append(action)
+    return states, next_states, actions 
+
+class CuriosityNetwork(nn.Module):
+
+    def __init__(self, obs_shape: Union[int, SequenceType], hidden_size_list: SequenceType, action_shape:int) -> None:
+        super(CuriosityNetwork, self).__init__()
+        if isinstance(obs_shape, int) or len(obs_shape) == 1:
+            self.feature= FCEncoder(obs_shape, hidden_size_list)
+        elif len(obs_shape) == 3:
+            self.feature= ConvEncoder(obs_shape, hidden_size_list)
+        else:
+            raise KeyError(
+                "not support obs_shape for pre-defined encoder: {}, please customize your own RND model".
+                format(obs_shape)
+            )
+        self.device = 'cpu' #torch.device('cuda' if use_cuda else 'cpu')
```

Comment:
Don't hardcode device here, pass `device` as the arguments of constructor method.